src/pages/workspace/workflows/ToggleSettingsOptionRow.tsx

```diff
@@ -0,0 +1,79 @@
+import React, {useEffect, useState} from 'react';
+import {View} from 'react-native';
+import type {ViewStyle} from 'react-native';
+import type {SvgProps} from 'react-native-svg';
+import Icon from '@components/Icon';
+import Switch from '@components/Switch';
+import Text from '@components/Text';
+import useThemeStyles from '@hooks/useThemeStyles';
+import useWindowDimensions from '@hooks/useWindowDimensions';
+
+type OptionType = {
+    icon: React.FC<SvgProps>;
+    title: string;
+    subtitle: string;
+    hasBeenToggled: boolean;
+    onToggle: (isEnabled: boolean) => void;
+    subMenuItems?: React.ReactNode;
+    isEndOptionRow?: boolean;
+};
+
+function ToggleSettingOptionRow({icon, title, subtitle, onToggle, subMenuItems, isEndOptionRow, hasBeenToggled}: OptionType) {
+    const [isEnabled, setIsEnabled] = useState(false);
```

Comment:
Why do we always default the state to false? Shouldn't it be initialized to the value on value stored in the policy object?

Reply:
The default value on initialization should be false, as it's the state by default.

With the `useEffect()` we check for the value of `hasBeenToggled`.

Why `hasBeenToggled`?
Because the values we need to check in the policy are specific and differ from the type of scenario we are at:
- Delayed submission: policy?.harvesting?.enabled
- Approvals enabled: policy?.isAutoApprovalEnabled

Plus, we don't pass a whole object to the other component which would be more performant too.